src/external/wannier90.jl

```diff
@@ -287,3 +289,79 @@ default_wannier_centres(n_wannier) = [rand(1, 3) for _ in 1:n_wannier]
     @timing "Wannierization" wannier90_jll.wannier90(exe -> run(Cmd(`$exe $prefix`; dir)))
     fileprefix
 end
+
+@doc raw"""
+    For a given set of bands ``\{Ψ_{nk}\}_{1\leq n \leq N_in}``
+    and a Wannier90 file containing unitary matrices (for disentanglement or
+    Wannierization) returns a set of band ``{Ψ_out_{nk}}_{1\leq n \leq N_out}``
+    with:
+    ``Ψ_out_{nk} = \sum\limits_{m=1}^{N_in} U_{mn}Ψ_{nk}``
+    ``N_in > N_out`` for disentanglement.
+    ``N_in = N_out`` for wannierization.
+"""
+function apply_U_matrices(file, ψ)
+    # Extract all Uk
+    @assert(isfile(file))
+    Uks = readdlm(file); num_kpts, num_ψout, num_ψin = Uks[2,1:3]
+    len_Uk = num_ψin*num_ψout
+
+    ψ_out = copy(ψ) # Better ideas than copy to inherit the type of ψ in ψ_out ?
+    for k in 1:num_kpts
+        # Locate the beginning of the matrix Uk written as a vector
+        i_Uk = 4 + (k-1)*(len_Uk+1)
+        Uk = reshape([x+im*y for (x,y) in eachrow(Uks[i_Uk:i_Uk+(len_Uk-1), 1:2])],
+                     num_ψin, num_ψout)
+        # Check that Uk is unitary
+        (norm(Uk'Uk - diagm(ones(num_ψout))) > 1e-8) && (@error("U matrix is not unitary"))
+        # Compute all ψ_out[k][:,n]
+        ψ_out_k = zero(ψ[k][:,1:num_ψout])
+        for n in 1:num_ψout
+            ψ_out_k[:,n] = ψ[k][:,1:num_ψin]*Uk[:,n]
+        end
+        ψ_out[k] = ψ_out_k
+    end
+    ψ_out
+end
+
+@doc raw"""
+    Wraps disentanglement (if needed) and wannierization to return Wanniers function
+    as a table ``W[k][G,n]``
+    The prefix in argument is the one returned by the `run_wannier90` routine above.
+    `ψ_unfold` is the array of Bloch waves from an "unfolded" self-consistent-field result.
```

Comment:
careful that we use unfolded in a (related but) different sense (unfold the symmetries of a BZ). Let's discuss this on slack

Reply:
ah no I see that you use it in the sense that we do in the symmetry.

Reply:
Yes I had trouble to find a good name for it. It is unfolded in the sense of symmetries but since we do not have any mention of the "folded" basis in this routine, maybe we can call it `\psi_nosym` or something equivalent ?